torch/nn/functional.py

```diff
@@ -1310,7 +1310,27 @@ def dropout2d(input: Tensor, p: float = 0.5, training: bool = True, inplace: boo
         return handle_torch_function(dropout2d, (input,), input, p=p, training=training, inplace=inplace)
     if p < 0.0 or p > 1.0:
         raise ValueError("dropout probability has to be between 0 and 1, " "but got {}".format(p))
-    return _VF.feature_dropout_(input, p, training) if inplace else _VF.feature_dropout(input, p, training)
+    inp_dim = input.dim()
+    msg = f"dropout2d: Input should be 3-D (unbatched) or 4-D (batched) but received {inp_dim}-D tensor"
+    assert inp_dim in (2, 3, 4), msg
+    if inp_dim == 2:
```

Comment:
@kshitij12345 Looks like some internal code is passing 5D inputs as well! How about this:
* Remove the assert entirely for now
* Throw the warning when encountering any input dim not in `(3, 4)`
* Analogous change for `Dropout3d`

Reply:
Done.